jsyntrax/src/main/java/org/atpfivt/jsyntrax/generators/SVGCanvasBuilder.java

```diff
@@ -46,26 +47,56 @@ public SVGCanvasBuilder withStyle(Style style) {
         return this;
     }
 
-    public SVGCanvasBuilder withUrlMap(Map<String, String> urlMap){
+    public SVGCanvasBuilder withUrlMap(Map<String, String> urlMap) {
         this.urlMap = urlMap;
         return this;
     }
 
+    public SVGCanvasBuilder withTitle(String title) {
+        this.title = title;
+        return this;
+    }
+
     public SVGCanvas generateSVG(Unit root) {
         this.canvas = new SVGCanvas(this.style, urlMap);
         parseDiagram(new Line(
                 new ArrayList<>(List.of(new Bullet(), root, new Bullet()))
         ), true);
-        return this.canvas;
-    }
 
+        if (title == null) {
+            return canvas;
+        }
 
-    public void addNodeStyle(NodeStyle ns) {
-        style.addNodeStyle(ns);
-    }
+        String tag = canvas.getCanvasTag();
+        Pair<Pair<Integer, Integer>, Pair<Integer, Integer>> bbox =
+                canvas.getBoundingBoxByTag(tag);
+
+        String titleTag = canvas.new_tag("x", "-title");
+        Element e = new TitleElement(title, style.title_font, "title_font", titleTag);
+        canvas.addElement(e);
+
+        if (style.title_pos.isLeft()) {
```

Comment:
OK, `left`/`middle`/`right` are exhaustive set of mutually exclusive alternatives, and so is `top`/`bottom`.

So we can use switch to cover them all and write something like this:

```java
        switch (style.title_pos) {
            case tl:
            case bl:
                canvas.moveByTag(titleTag, 2 * style.padding, 0);
                break;
            case tm:
            case bm:
                //don't forget breaks!

        }
        
        switch (style.title_pos){
            case tl:
            case tm:
            case tr:
                ///
                break;
                
            case bl:
            case br:
            case bm:
                //
                
        }
```

and there will be no need for all the getters